src/haarfeature.cpp

```diff
@@ -0,0 +1,279 @@
+#include <vector>
+#include <algorithm>
+#include <random>
+#include <utility>
+#include "./haarfeature.h"
+
+namespace myHaar {
+
+Point::Point(int new_x, int new_y) : x(new_x), y(new_y) {}
+
+int Point::getX() const {
+  return Point::x;
+}
+
+int Point::getY() const {
+  return Point::y;
+}
+
+Point& Point::operator=(const Point new_point) {
+  Point::x = new_point.x;
+  Point::y = new_point.y;
+  return *this;
+}
+
+Size::Size(int new_width, int new_height) :
+           width(new_width), height(new_height) {}
+
+int Size::getWidth() const {
+  return Size::width;
+}
+
+int Size::getHeight() const {
+  return Size::height;
+}
+
+Size& Size::operator*(int value) {
+  Size::width *= value;
+  Size::height *= value;
+  return *this;
+}
+
+Size& Size::operator=(const Size new_size) {
+  Size::width = new_size.width;
+  Size::height = new_size.height;
+  return *this;
+}
+
+int verticalRatio(haarFeatureType type) {
```

Comment:
lets use the rule "one class -- one file"
and move out the outside functions to a separate file (or make them a part of class)